crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -36,13 +36,27 @@ impl Engine for WasmtimeEngine {
 
         log::info!("wasi context ready");
         let WasiEntrypoint { path, func } = ctx.wasi_entrypoint();
-        let path = path
-            .resolve_in_path_or_cwd()
-            .next()
-            .context("module not found")?;
 
-        log::info!("loading module from file {path:?}");
-        let module = Module::from_file(&self.engine, &path)?;
+        let module = match ctx.oci_artifacts() {
```

Comment:
Might be nice to have some shared utilities to dry up the repeated code blocks.

Reply:
open to ideas... I not sure the best way to do this and maybe it's to early? it is likely the processing will change once we start to move over to preview2, which some runtimes will support and some won't?

Another thing to consider is implementing something like pre-compliation (#44) and how each runtime composes components together.  Should that be done before and only the compiled/composed module passed in?